deltacat/aws/constants.py

```diff
@@ -1,10 +1,26 @@
-from typing import List, Set
+import botocore
+from typing import Set
 
 from deltacat.utils.common import env_integer, env_string
 
 DAFT_MAX_S3_CONNECTIONS_PER_FILE = env_integer("DAFT_MAX_S3_CONNECTIONS_PER_FILE", 8)
 BOTO_MAX_RETRIES = env_integer("BOTO_MAX_RETRIES", 5)
-TIMEOUT_ERROR_CODES: List[str] = ["ReadTimeoutError", "ConnectTimeoutError"]
-RETRYABLE_PUT_OBJECT_ERROR_CODES: Set[str] = {"Throttling", "SlowDown"}
+TIMEOUT_ERROR_CODES: Set[str] = {"ReadTimeoutError", "ConnectTimeoutError"}
+THROTTLING_ERROR_CODES: Set[str] = {"Throttling", "SlowDown"}
+RETRYABLE_TRANSIENT_ERRORS = (
+    ConnectionError,
+    TimeoutError,
+    OSError,
+    botocore.exceptions.ConnectionError,
+    botocore.exceptions.HTTPClientError,
+    botocore.exceptions.NoCredentialsError,
+    botocore.exceptions.ConnectTimeoutError,
+    botocore.exceptions.ReadTimeoutError,
```

Comment:
Can we also enrich it by retrieving exceptions using exception env variables? For example, we can specify `botocore.exceptions.ReadTimeoutError` as value of env variable and the code here imports it at runtime using importlib library. I wanted to do this, but never got time to do it. This feature may be way more useful now than ever because, you don't need to create a new deltacat version if you find that you missed any transient error.

Reply:
Perhaps you could raise this as a separate feature request / PR? I feel that this is a more nuanced use case and we'll want to test that feature more carefully (e.g. if you try to specify an exception class that doesn't exist or is no longer used in the dependencies, we should ignore it gracefully)

Reply:
it can be a separate PR but we agreed the new deltacat release should go with this feature to improve release efficiency when we find a new error we want to retry.